shell/renderer/api/electron_api_context_bridge.cc

```diff
@@ -810,13 +857,233 @@ bool OverrideGlobalPropertyFromIsolatedWorld(
+    return object_value;
+  }
+
+  // Attempt to clone value.
+  v8::MaybeLocal<v8::Value> maybe_result;
+  {
+    v8::TryCatch try_catch(isolate);
+    context_bridge::ObjectCache object_cache;
+    maybe_result =
+        PassValueToOtherContext(creation_context, target_context, object_value,
+                                creation_context->Global(), &object_cache,
+                                false, 0, BridgeErrorTarget::kSource);
+    if (try_catch.HasCaught()) {
+      v8::String::Utf8Value utf8(isolate, try_catch.Exception());
+      error_message = *utf8 ? *utf8 : "Unknown error cloning result";
+    }
+  }
+
+  return maybe_result;
+}
+
+// Serialize script to be executed in the given world.
+v8::Local<v8::Value> ExecuteInWorld(v8::Isolate* isolate,
+                                    const int world_id,
+                                    gin_helper::Arguments* args) {
+  // Get execution script argument
+  gin_helper::Dictionary exec_script;
+  if (args->Length() >= 1 && !args->GetNext(&exec_script)) {
+    gin_helper::ErrorThrower(args->isolate()).ThrowError("Invalid script");
+    return v8::Local<v8::Value>();
+  }
+
+  // Get "func" from execution script
+  v8::Local<v8::Function> func;
+  if (!exec_script.Get("func", &func)) {
+    gin_helper::ErrorThrower(isolate).ThrowError(
+        "Function 'func' is required in script");
+    return v8::Local<v8::Value>();
+  }
+
+  // Get optional "args" from execution script
+  v8::Local<v8::Array> args_array;
+  v8::Local<v8::Value> args_value;
+  if (exec_script.Get("args", &args_value)) {
+    if (!args_value->IsArray()) {
+      gin_helper::ErrorThrower(isolate).ThrowError("'args' must be an array");
+      return v8::Local<v8::Value>();
+    }
+    args_array = args_value.As<v8::Array>();
+  } else {
+    args_array = v8::Array::New(isolate);
+  }
+
+  // Serialize the function
+  v8::Local<v8::String> serialized_func;
+  if (!func->FunctionProtoToString(isolate->GetCurrentContext())
+           .ToLocal(&serialized_func)) {
+    gin_helper::ErrorThrower(isolate).ThrowError(
+        "Failed to serialize function");
+    return v8::Local<v8::Value>();
+  }
+  v8::String::Utf8Value serialized_func_utf8(isolate, serialized_func);
+  std::string func_str(*serialized_func_utf8, serialized_func_utf8.length());
+
+  v8::Local<v8::Context> source_context = isolate->GetCurrentContext();
+  v8::Context::Scope source_scope(source_context);
+
+  // Get the target context
+  v8::MaybeLocal<v8::Context> maybe_target_context =
+      GetTargetContext(isolate, world_id);
+  v8::Local<v8::Context> target_context;
+  if (!maybe_target_context.ToLocal(&target_context)) {
+    isolate->ThrowException(v8::Exception::Error(gin::StringToV8(
+        isolate,
+        base::StringPrintf("Failed to get context for world %d", world_id))));
+    return v8::Local<v8::Value>();
+  }
+
+  // Compile the script
+  v8::MaybeLocal<v8::Script> maybe_compiled_script;
```

Comment:
We should be compiling the script in the `target_scope`, to make this very clear the target scope should be created before making this empty maybelocal IMO. These scopes need to stand up to refactors, moves, etc. so ensuring that the block `{}` are very clear is important

Reply:
> These scopes need to stand up to refactors, moves, etc.

I like and agree with this point, but I'm confused on how to ensure this given the code being commented on.

The compiled script will need to be referenced later on in the function. The best I'm able to interpret this would be something to the effect of
```cpp
v8::Local<v8::Script> compiled_script;
{
  v8::Context::Scope target_scope(target_context);
  // compile the script and assign `compiled_script` before exiting the block
}
```

Let me know if this is on the right track.

Reply:
Latest changes should have better block-level scoping. Let me know if there's more that can be done.